toolkit/tools/precacher/precacher.go

```diff
@@ -348,9 +348,13 @@ func monitorProgress(total int, results chan downloadResult, doneChannel chan st
 // network operations semaphore to minimize the time spent holding it.
 func precachePackage(pkg *repocloner.RepoPackage, packagesAvailableFromRepos map[string]string, outDir string, wg *sync.WaitGroup, results chan<- downloadResult, netOpsSemaphore chan struct{}) {
 	const (
-		downloadRetryAttempts = 2
+		// With 5 attempts, initial delay of 1 second, and a backoff factor of 2.0 the total time spent retrying will be
+		// ~30 seconds.
+		downloadRetryAttempts = 5
+		failureBackoffBase    = 2.0
 		downloadRetryDuration = time.Second
 	)
+	var noCancel chan struct{} = nil
```

Comment:
Nit-pick: can this be a `const` as well, since it looks like we can make constants typed?

Reply:
No, channel is considered non-constant and must be a var (I tried).